csharp_dotnetcore/3.Welcome-Users/WelcomeUserBot.cs

```diff
@@ -0,0 +1,146 @@
+                }
+            }
+
+            // greet when users are added to the conversation
+            else if (turnContext.Activity.Type == ActivityTypes.ConversationUpdate)
+            {
+                if (turnContext.Activity.MembersAdded.Any())
+                {
+                    // go over all new memebers, greet anyone that was not the target (recipient) of this message
+                    // the 'bot' is the recipient for events from the channel
+                    foreach (var member in turnContext.Activity.MembersAdded)
+                    {
+                        // turnContext.Activity.MembersAdded == turnContext.Activity.Recipient.Id indicates the
+                        // bot was added to the conversation
+                        if (member.Id != turnContext.Activity.Recipient.Id)
+                        {
+                            await turnContext.SendActivityAsync($"Hi there - {member.Name}. Welcome to the \'Welcome User\' Bot. This bot will introduce you to Welcoming and greeting users.", cancellationToken: cancellationToken);
+                            await turnContext.SendActivityAsync($"You are seeing this message, because the bot recieved atleast one \'ConversationUpdate\' event, indicating you (and possibly others) joined a conversation. If you are using the emulator, pressing the \'Start Over\' trigger thise event. \'ConversationUpdate\' event depends on the channel your bot is running. You can read more information https://aka.ms/about-botframewor-welcome-user", cancellationToken: cancellationToken);
+                        }
+                    }
+                }
+            }
+
+            // default behaivor for all other type of events.
+            else
+            {
+                var ev = turnContext.Activity.AsEventActivity();
+                await turnContext.SendActivityAsync($"Received event: {ev.Name}");
+            }
+        }
+
+        /// <summary>
+        /// sends an adaptive card introduction.
+        /// </summary>
+        /// <param name="turnContext">A <see cref="ITurnContext"/> containing all the data needed
+        /// for processing this conversation turn. </param>
+        /// <param name="cancellationToken">(Optional) A <see cref="CancellationToken"/> that can be used by other objects
+        /// or threads to receive notice of cancellation.</param>
+        /// <returns>A task that represents the work queued to execute.</returns>
+        private static async Task SendIntroCardAsync(ITurnContext turnContext, CancellationToken cancellationToken)
+        {
+            var response = turnContext.Activity.CreateReply("This is an hello world intro to Adaptive Cards");
+
+            var introCard = File.ReadAllText(@".\Resources\IntroCard.json");
+
+            response.Attachments = new List<Attachment>
+            {
+                new Attachment()
+                {
+                    ContentType = "application/vnd.microsoft.card.adaptive",
```

Comment:
There is a constant for this. Replace this string with the constant. If that constant doesn't exist, please add it in the right place...